### PR TITLE
CRUD for client and question #7

### PR DESCRIPTION
**Pull request by CortesV**

### CODE REVIEW

File: src/main/java/com/survey/softbistro/client/manage/components/interfaces/IClient.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	public Response saveClient(Client client);
+
+	/**
+	 * Delete client from database by email of client
+	 * 
+	 * @param email
+	 *            email - email of client
+	 * @return return - status of execution this method
+	 */
+	public Response deleteClient(String email);
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/client/manage/components/interfaces/IClient.java]: use id

File: src/main/java/com/survey/softbistro/client/manage/service/ClientService.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+/**
+ * Service for CRUD of Client
+ * @author cortes
+ *
+ */
+@Service
+public class ClientService {
+
+	@Autowired
+	IClient iClient;
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/client/manage/service/ClientService.java]: private

File: src/main/java/com/survey/softbistro/client/manage/controller/ClientController.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * Controller for CRUD of Client
+ * @author cortes
+ *
+ */
+@RestController
+@RequestMapping(value = "/rest/survey/v1/client")
+public class ClientController {
+
+	@Autowired
+	ClientService clientService;
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/client/manage/controller/ClientController.java]: private

File: src/main/java/com/survey/softbistro/client/manage/components/service/ClientDao.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+	/**
+	 * Class for geting clients from database
+	 * 
+	 * @author cortes
+	 *
+	 */
+	public class WorkingWithRowMap implements RowMapper<Client> {
+
+		@Override
+		public Client mapRow(ResultSet resultSet, int rowNum) throws SQLException {
```
Comment (af150416) [inline src/main/java/com/survey/softbistro/client/manage/components/service/ClientDao.java]: Why not to use BeanPropertyRowMapper ?

File: src/main/java/com/survey/softbistro/question/controller/QuestionController.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ * 
+ * @author cortes
+ *
+ */
+@RestController
+@RequestMapping(value = "/rest/survey/v1/question")
+public class QuestionController {
+
+	@Autowired
+	QuestionService questionService;
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/question/controller/QuestionController.java]: private

File: src/main/java/com/survey/softbistro/question/service/QuestionService.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+import com.survey.softbistro.question.components.entity.Question;
+import com.survey.softbistro.question.components.interfaces.IQuestion;
+import com.survey.softbistro.response.Response;
+
+@Service
+public class QuestionService {
+
+	@Autowired
+	IQuestion iQuestion;
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/question/service/QuestionService.java]: private

File: src/main/java/com/survey/softbistro/question/service/QuestionService.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
+package com.survey.softbistro.question.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.survey.softbistro.question.components.entity.Question;
+import com.survey.softbistro.question.components.interfaces.IQuestion;
+import com.survey.softbistro.response.Response;
+
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/question/service/QuestionService.java]: java doc

File: src/main/java/com/survey/softbistro/startapp/SurveySoftBistroApplication.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.ComponentScan.Filter;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.FilterType;
+
+@SpringBootApplication
+//@ComponentScan("com.survey.softbistro")
```
Comment (ZVIproject) [inline src/main/java/com/survey/softbistro/startapp/SurveySoftBistroApplication.java]: why commit

File: src/main/java/com/survey/softbistro/client/manage/service/ClientService.java (hunk 2 above)
Comment (CortesV) [inline src/main/java/com/survey/softbistro/client/manage/service/ClientService.java]: done

File: src/main/java/com/survey/softbistro/client/manage/controller/ClientController.java (hunk 3 above)
Comment (CortesV) [inline src/main/java/com/survey/softbistro/client/manage/controller/ClientController.java]: done

File: src/main/java/com/survey/softbistro/question/controller/QuestionController.java (hunk 5 above)
Comment (CortesV) [inline src/main/java/com/survey/softbistro/question/controller/QuestionController.java]: done

File: src/main/java/com/survey/softbistro/question/service/QuestionService.java (hunk 6 above)
Comment (CortesV) [inline src/main/java/com/survey/softbistro/question/service/QuestionService.java]: done

File: src/main/java/com/survey/softbistro/question/service/QuestionService.java (hunk 7 above)
Comment (CortesV) [inline src/main/java/com/survey/softbistro/question/service/QuestionService.java]: done

File: src/main/java/com/survey/softbistro/startapp/SurveySoftBistroApplication.java (hunk 8 above)
Comment (CortesV) [inline src/main/java/com/survey/softbistro/startapp/SurveySoftBistroApplication.java]: done

File: src/main/java/com/survey/softbistro/question/components/service/QuestionDao.java [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+	 * Find question in database by id of question
+	 * 
+	 * @param id
+	 *            id - id of question
+	 * @return return - all information about question
+	 */
+	@Override
+	public Response findQuestionById(Long id) {
+
+		Question question = new Question();
```
Comment (af150416) [inline src/main/java/com/survey/softbistro/question/components/service/QuestionDao.java]: Why do you use new operator ?

**Pull request merged**